music_assistant/server/controllers/music.py

```diff
@@ -157,6 +162,39 @@ async def search(
         :param media_types: A list of media_types to include.
         :param limit: number of items to return in the search (per type).
         """
+        # Check if the search query is a streaming provider public shareable URL
+        try:
+            media_type, provider_instance_id_or_domain, item_id = parse_uri(
+                search_query, validate_id=True
+            )
+        except InvalidProviderURI as err:
+            self.logger.warning("%s", str(err))
+            return SearchResults()
+        except MusicAssistantError:
+            pass
+        else:
+            if provider_instance_id_or_domain in PROVIDERS_WITH_SHAREABLE_URLS:
```

Comment:
You could consider moving this block below the parse_uri and just do everything in one GO - catch InvalidProviderURI and let that pass (as it means there was no uri passed) - catch MusicAssistantError and log that one.

Reply:
Hmm yeah need to give this some thought.  If we do as I suggest above and create another exception class, we're now catching three with different handlers.  Moving this block inside the `try:` will look prettier, but will it aid readability when we come back to look at it later and now there's multiple things going on in the `try:` block and we need to work out which thing can raise which exception?  Especially given `get_item` is a multi-pathed beast.  It's up to you though, I'm happy to go with whatever you feel is best ☺️

Reply:
Let's go with an iterative approach, fix the issue now and make it prettier later.